tests/zfs-tests/tests/perf/fio/random_readwrite.fio

```diff
@@ -30,7 +31,10 @@ ioengine=psync
 sync=${SYNC_TYPE}
 direct=${DIRECT}
 numjobs=${NUMJOBS}
+randseed=1234
 buffer_compress_percentage=66
-buffer_compress_chunk=4096
+refill_buffers
+buffer_pattern=0xdeadbeef
+buffer_compress_chunk=0
```

Comment:
Doesn't using such as short `buffer_pattern` result in unrealistically good compression?  What does ZFS report the actually compression factor to be?  This would be a problem for this kind of performance test which is trying to create semi-realistic data.

Reply:
According to the documentation:

>  The fixed pattern is either zeros, or the pattern specified by buffer_pattern. If the buffer_pattern option is used, it might skew the compression ratio slightly

However, doing some more reading, I think I can drop the `refill_buffers` portion:

>  Setting buffer_compress_percentage to a value other than 100 will also enable refill_buffers in order to reduce the likelihood that adjacent blocks are so similar that they over compress when seen together

Reply:
Refill_buffers removed (as stated above)

Reply:
Got it, OK thanks for verifying the compressibility percentage.  Keeping this the same is desirable since I believe @jwk404 and others have historical data for these performance tests and we want to be able to continue to compare against those results.

Reply:
Using the old and new `mkfiles.fio` files, I see a marked difference in compression ratio (with lz4) with these changes. The ratio goes from 2.87x to 1.50x. I'm open to changes here, but would prefer the ratio remains closer to what it was.

Reply:
@jwk404 1.50x actually IS the 66(%) you are asking fio for, so your 2.87 is basically a bug. 
That being said: 1.50x is too low, I agree.

Taking it to the limit I can push it to about 1.97... Would that suffice?
If you want more, the solution is basically using something that feeds plaintext, which is not fio.

In any case, the current 2.87 is just zero suppression, so doesn;t mean anything...

Reply:
I'm not able to push it byond 1.97 on `buffer_compress_percentage=50`
Because of the following bug:
https://github.com/axboe/fio/issues/891

If I was able to select a bigger portion of the buffer to be compressable, I would even been able to keep your 2.87 in tact. Heck, I would even be able to fill the buffer with an enwik dataset...

Reply:
Parameters that produce ~3x compression were chosen deliberately, because that roughly matches the compression we see on the datasets these tests were written for. It's not a bug.

I think a workable solution would be to modify the files such that the defaults remain as they are, but can be modified with environment variables for users who want to test other types of datasets and compression algorithms.

Reply:
@jwk404 I understand why you have choosen to do it this way and yes, 3x compression would be preferable.

I'm willing to modify my PR to remain within your guideline of 3x compression.

So: I'm going to work on this as soon as above bug is solved. If that bug is solved I would be able to give you your 3x compression.

Reply:
@Ornias1993 That sounds great. Thanks for doing the extra work (especially filing the fio issue).

Reply:
No problem and sorry for my misunderstanding earlier...

Reply:
@jwk404 
I've done some research, my initial understanding was that the buffer_pattern was leading to this lower compressability.

My new research indicates that buffer_pattern had nothing to do with it.

The cause of the lower compression was the buffer_compress_chunk.
Simply put: When using 4096, multiple chunks fit into one 1024k block, this leads to a situation where it's easier to compress than using blocksize-sized chunks (chunk = 0).

To fix it i'll make sure chunk size is set-able by variable and to be sure i'll do the same with the compress percentage.